Index: llvm-project/mlir/test/lit.cfg.py
===================================================================
--- llvm-project.orig/mlir/test/lit.cfg.py	2023-12-19 16:44:21.448594000 -0600
+++ llvm-project/mlir/test/lit.cfg.py	2023-12-20 17:59:45.280300483 -0600
@@ -52,11 +52,15 @@
 # Correctly handles the platforms shared library directory and naming conventions.
 def add_runtime(name):
     path = ""
+    print(f"config.llvm_shlib_dir:{config.llvm_shlib_dir}")
+    print(f"config.llvm_shlib_ext:{config.llvm_shlib_ext}")
     for prefix in ["", "lib"]:
         path = os.path.join(
-            config.llvm_shlib_dir, f"{prefix}{name}{config.llvm_shlib_ext}"
+            config.mlir_tools_dir, "../lib", f"{prefix}{name}{config.llvm_shlib_ext}"
         )
+        print(f"path:{path}")
         if os.path.isfile(path):
+            print(f"file found:{path}")
             break
     return ToolSubst(f"%{name}", path)
 
@@ -80,6 +84,12 @@
 # Tweak the PATH to include the tools dir.
 llvm_config.with_environment("PATH", config.mlir_tools_dir, append_path=True)
 llvm_config.with_environment("PATH", config.llvm_tools_dir, append_path=True)
+print(f"config.mlir_tools_dir:{config.mlir_tools_dir}")
+print(f"config.llvm_tools_dir:{config.llvm_tools_dir}")
+print(f"config.mlir_lib_dir:{config.mlir_lib_dir}")
+print(f"config.llvm_shlib_dir:{config.llvm_shlib_dir}")
+print(f"config.mlir_src_root:{config.mlir_src_root}")
+
 
 tool_dirs = [config.mlir_tools_dir, config.llvm_tools_dir]
 tools = [
Index: llvm-project/mlir/test/CMakeLists.txt
===================================================================
--- llvm-project.orig/mlir/test/CMakeLists.txt	2023-12-19 16:43:03.439912000 -0600
+++ llvm-project/mlir/test/CMakeLists.txt	2023-12-20 16:44:33.359165730 -0600
@@ -12,6 +12,8 @@
 
 # Passed to lit.site.cfg.py.in to set up the path where to find libraries.
 set(MLIR_LIB_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+# set(MLIR_LIB_DIR "${SRC_DIR}/build/lib")
+# set(SHLIBDIR "${SRC_DIR}/build/lib")
 
 if (MLIR_INCLUDE_INTEGRATION_TESTS)
   set(INTEL_SDE_EXECUTABLE "" CACHE STRING
Index: llvm-project/mlir/test/Integration/lit.local.cfg
===================================================================
--- llvm-project.orig/mlir/test/Integration/lit.local.cfg	2023-12-20 16:04:28.409076600 -0600
+++ llvm-project/mlir/test/Integration/lit.local.cfg	2023-12-20 16:05:27.145124218 -0600
@@ -20,6 +20,7 @@
             config.arm_emulator_utils_lib_dir or config.mlir_lib_dir,
         )
     )
+    print(f"config.mlir_native_utils_lib_dir:{config.mlir_native_utils_lib_dir}")
 
     if config.arm_emulator_executable:
         if config.arm_emulator_lli_executable:
